### ISSUE TITLE
Prices reset to 0$ after last update

### ISSUE THREAD

**Issue by Bloodlex** [alex9849/advanced-region-market#127]
Hey,

After the last 3.4.2 update, many of my regions changed price to 0$ (mostly poligonal ones if that is important), although they had been created previously with autoprice. And creating sign once again with autoprice still doesn't change it.

For now I rolled back to previous version.

Do you have any idea why it cold be so?

Thanks for your help.

Best regards,
Bloodlex

**Comment by alex9849** [comment]
Not really. Maybe the calculations for volumes break in the newest version. I'll need to check that :)

**Comment by Bloodlex** [comment]
> Not really. Maybe the calculations for volumes break in the newest version. I'll need to check that :)

I'd be grateful if you could, thanks a lot 🙂

**Comment by TheClowner** [comment]
I'm running into this issue myself now, any poly regions just don't calculate a volume - probably since worldguard's .volume() method for poly regions just returns 0 - https://github.com/EngineHub/WorldGuard/blob/20044a72d3f5bd3d6a6a2f1e00e0b7307789bb48/worldguard-core/src/main/java/com/sk89q/worldguard/protection/regions/ProtectedPolygonalRegion.java#L186 

I guess maybe ARM used to calculate the volume itself?

**Comment by Emix33** [comment]
This "regression"? seems to have occured in commit 2e9d06ec , Before this commit Arm did indeed workaround the worldguard bug by calculating the volume for polygonal regions itself. I dont know why this was changed, or if it was intentional.

**Comment by Bloodlex** [comment]
> This "regression"? seems to have occured in commit [2e9d06e](https://github.com/alex9849/advanced-region-market/commit/2e9d06ec08e2631c470a1c5abee972c2be2a5fc9) , Before this commit Arm did indeed workaround the worldguard bug by calculating the volume for polygonal regions itself. I dont know why this was changed, or if it was intentional.

Yes, it worked before.
